app/views/search/search.html.erb

```diff
@@ -0,0 +1,180 @@
+<%= provide(:title, "Search") %>
+<script type='text/javascript' src='https://maps.googleapis.com/maps/api/js?key=AIzaSyANRR2kNcJ85_61PYyROVKQrRjRtqhpEUs&callback=initialize'></script>
+<script type='text/javascript'>
+    document.addEventListener("DOMContentLoaded", function() {
+        var location = [];
+        <% @listing_list.each do |pin| %>
+            location.push({lat: Number("<%= pin.latitude %>"), lng: Number("<%= pin.longitude %>")});
+        <% end %>
+        
+        var infoWindow = [];
+        <% @listing_list.each do |info| %>
+            infoWindow.push(new google.maps.InfoWindow({
+                content: '<div id="content"><%= info.price %></div>'
+            }));
+        <% end %>
+
+        var map = new google.maps.Map(document.getElementById('map'),{
+            center: location[0],
+            zoom: 14
+        });
+
+        var marker = [];
+        for(var i=0; i<location.length; i++){
+            marker[i] = new google.maps.Marker({
+                position: location[i],
+                map: map
+            });
+        }
+        for(var i=0; i<location.length; i++){
+            infoWindow[i].open(map, marker[i]);
+        }
+    });
+</script>
+
+<div class="offset-4 col-5 mt-5">
+    <button type="button" class="btn btn-dark w-25" data-toggle="collapse" data-target="#collapseExample" aria-expanded="false" aria-controls="collapseExample">
+        Filter
+    </button>
+</div>
+<div class="collapse col-8 mt-4" id="collapseExample">
+    <div class="border p-3">
+        <form>
+            <div class="form-group">
+                <label for="formControlRange">Example Range input</label>
+                <input type="range" class="form-control-range" id="formControlRange">
+            </div>
+        </form>
+        <form class="form-inline">
+            <h5>Price Range</h5>
+            <div class="form-group ml-5">
+                <%= search_form_for @search, url: search_path do |f| %>
+                    <%= f.label :price, "Min Price" %>
+                    <%= f.search_field :price_cont, class: "form-control mr-sm-2 ml-3" %>
+                    <%= f.label :price, "Max Price", class: "ml-5" %>
+                    <%= f.search_field :price_cont, class: "form-control mr-sm-2 ml-3" %>
```

Comment:
make sure this function as a range of value
range must be using _greater than or equal to_ and _less than or equal to_ 
you can reference here for the list of ransack parameters
https://github.com/activerecord-hackery/ransack#search-matchers